packages/klighd-core/src/options/render-options-registry.ts

```diff
@@ -16,18 +16,147 @@ import { injectable } from "inversify";
 import { Action, ICommand } from "sprotty";
 import { Registry } from "../base/registry";
 import { SetRenderOptionAction } from "./actions";
-import { RenderOption, TransformationOptionType } from "./option-models";
+import { RangeOption, RenderOption, TransformationOptionType } from "./option-models";
 
 export class ShowConstraintOption implements RenderOption {
     static readonly ID: string = "show-constraints";
     static readonly NAME: string = "Show Constraint";
     readonly id: string = ShowConstraintOption.ID;
     readonly name: string = ShowConstraintOption.NAME;
     readonly type: TransformationOptionType = TransformationOptionType.CHECK;
+    readonly updateNeeded: boolean = true;
     readonly initialValue: boolean = false;
     currentValue = false;
 }
 
+/**
+ * Boolean option to enable and disable the smart zoom feature.
+ * This corresponds to the automatic expansion and collapsing of regions and states
+ * as well as limiting visible elements.
+ */
+ export class UseSmartZoom implements RenderOption {
+    static readonly ID: string = 'use-smart-zoom'
+    static readonly NAME: string = 'Smart Zoom'
+    readonly id: string = UseSmartZoom.ID
+    readonly name: string = UseSmartZoom.NAME
+    readonly type: TransformationOptionType = TransformationOptionType.CHECK
+    readonly updateNeeded: boolean = false
+    readonly initialValue: boolean = true
+    currentValue = true
+}
+
+/**
+ * Threshold for expansion as well as collapse of states.
+ * Corresponds to the regions size compared to the current viewport.
+ */
+ export class ExpandCollapseThreshold implements RangeOption {
+    static readonly ID: string = 'expand-collapse-threshold'
+    static readonly NAME: string = 'Expand Collapse Threshold'
+    readonly id: string = ExpandCollapseThreshold.ID
+    readonly name: string = ExpandCollapseThreshold.NAME
+    readonly type: TransformationOptionType = TransformationOptionType.RANGE
+    readonly updateNeeded: boolean = false
```

Comment:
Why is this field added if it is always false? Especially this option seems to benefit if changing the threshold would also live update the diagram accordingly.

Reply:
* This is `true` for the `ShowConstraintOption` above in Line 27.
*  Client Side the Diagram should still be updated (TODO), but this does not require the LS to regenerate the model

Reply:
this field has now been removed again as you are right that it is not necessary as it can be false for all RenderOptions